lib/index.js

```diff
@@ -369,18 +369,23 @@ exports.deepEqual = function (obj, ref, options, seen) {
 
 // Remove duplicate items from array
 
-exports.unique = function (array, key) {
-
-    const index = {};
-    const result = [];
-
-    for (let i = 0; i < array.length; ++i) {
-        const id = (key ? array[i][key] : array[i]);
-        if (index[id] !== true) {
-
-            result.push(array[i]);
-            index[id] = true;
-        }
+exports.unique = (array, key) => {
+
+    let result;
+    if (key) {
+        result = [];
+        const index = new Set();
+        array.forEach((item) => {
+
+            const identifier = item[key];
+            if (!index.has(identifier)) {
```

Comment:
i wonder if it would be more efficient to build the set without checking `.has` (since sets don't allow duplicate members anyway), and then returning `Array.from(theSet)` like we do below. building both a set and an array at the same time seems wasteful.